src/Windowing/Extensions/Silk.NET.Windowing.Extensions.WebGPU/WebGPUWindow.cs

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Silk.NET.Core.Contexts;
+using Silk.NET.WebGPU;
+using Silk.NET.Windowing;
+
+namespace Silk.NET.Windowing.Extensions.WebGPU;
+/// <inheritdoc/>
+
+public static class WebGPUWindow
+{
+    /// <summary>
+    /// Creates a WebGPU surface from a window.
+    /// </summary>
+    /// <param name="wgpu">A WebGPU instance</param>
+    /// <param name="view">The window to get the surface from</param>
+    /// <returns>A newly created Surface*/</returns>
+    /// <exception cref="NotSupportedException">Throws when an unsupported platform is detected.</exception>
+    public static unsafe Surface* CreateWebGPUSurface(this IView view, NET.WebGPU.WebGPU wgpu, Instance* instance = null)
```

Comment:
Is there any reason why this can't be in Silk.NET.WebGPU? This method can be implemented entirely using Silk.NET.Core types which is available for free in Silk.NET.WebGPU.

This is a precedent that has been set in OpenXR with `GraphicsBinding`. Generally we don't add silk-to-silk extensions given we're the vendor of both sides (this is what Silk.NET.Core is designed to be, an interchange library to make our libraries integrate better together)

`IView` becomes `INativeWindowSource` which will still make the extension method accessible in the same way.